fc/mef.py

```diff
@@ -539,54 +261,67 @@ def sc_fun(p,x):
         return np.exp(p[0] * x + p[1])
     
     # Fit parameters
-    err_par = lambda p: err_fun(p, peaks_ch, peaks_mef)
+    err_par = lambda p: err_fun(p, fl_channel, fl_mef)
     res = minimize(err_par, params, options = {'gtol': 1e-6})
 
     # Separate parameters
-    sc_params = res.x
+    beads_params = res.x
 
     # Beads model function
-    sc_beads = lambda x: fit_fun(sc_params, x)
+    beads_model = lambda x: fit_fun(beads_params, x)
 
     # Standard curve function
-    sc = lambda x: sc_fun(sc_params, x)
+    std_crv = lambda x: sc_fun(beads_params, x)
     
-    return (sc, sc_beads, sc_params)
-
-def plot_standard_curve(peaks_ch, 
-                        peaks_mef,
-                        sc_beads,
-                        sc_abs,
-                        xlim = (0., 1023.),
-                        ylim = (1, 1e8)):
+    return (std_crv, beads_model, beads_params)
```

Comment:
For `fit_beads_autofluorescence()`, I made some changes after the fork a while back which I think may still be relevant:
* initialized fit with log line (from JS3xton/fc@96a311d1e0b04b6a26cfc104d575f468d28dbc2a):

    ```python
    if kp[0] == 0:
        # If lowest peak is a blank, remove it for the log-y line fit to
        # avoid trying to fit a line to log(0) = -Inf
        p = np.polyfit(fp[1:], np.log(kp[1:]), 1)
    else:
        p = np.polyfit(fp, np.log(kp), 1)

    params[0] = p[0]
    params[1] = p[1]
    ```
    I think this is generally more robust than what @thoreusc originally coded up which just fits the top two peaks to a line. That method is much more sensitive to top two peak values. This method is like fitting a simpler model first (log line with no fluorescence term which happens to have a closed form solution), and then using that to initialized our slightly more complicated model.
* changed `params[2]` to initialize based on the lowest peak in `fl_channel`. From JS3xton/fc@0dc3babb447931e8a860c75b7776f8752a4f7400:

    ```python
    params[2] = np.exp((params[0] * fp[0]) + params[1])
    ```
    This basically just solves the bead model for `fl_mef_auto` using the already calculated slope and intercept in conjunction with the first fit peak ("`fp`"; maybe should be `min(fp)` instead of `fp[0]`, as the intention is really the smallest peak value?). This basically initializes `bead_auto` with the value corresponding to the lowest subpopulation summarizing statistic (median here). I think this is a much better / more intuitive initialization than the arbitrary `100` that @thoreusc initially chose.
* I used the linear algebra `norm` to calculate my error function (JS3xton/fc@0dc3babb447931e8a860c75b7776f8752a4f7400):

    ```python
    res = scipy.optimize.minimize(
        fun=lambda p,x,y: np.linalg.norm(np.log(y + p[2]) - ((p[0] * x) + p[1])),
        x0=params,
        args=(fp,kp))
    ```
    I think I've always seen norm used in optimization texts. I believe the current implementation differs from norm because it lacks a final square root. While I don't think that changes the extrema, it might change where on the slope up to / down to the extrema the optimization algorithm stops.
* I see you return a lot of lambda functions. Is this a problem?: http://docs.python-guide.org/en/latest/writing/gotchas/#late-binding-closures  I think I tried to address this in my own version here (JS3xton/fc@d616c36cf2239c09954cd2a3969c85eb0ec06dfc):

    ```python
        # Python binds functions as late a possible, so all of the standards
        # curves would behave the same if we just returned standards_curve
        # (they would all rely upon the final resting params values after the
        # loop terminated). functools.partial addresses this problem by
        # binding some of the function parameters at definition time, not
        # call time.
        sc.append(functools.partial(standards_curve, params=params))
    ```
    You kind of have 2 layers of functions here (ones explicitly defined and then lambda functions) which is a little confusing, so I'm not actually sure what happens. Seems like your 2nd layer of lambda functions is equivalent to the partial functions I was using? Does that also solve the problem? I'm not sure.